### PR TITLE
Structure change

### PR DESCRIPTION
**Pull request by imjalpreet**
@oparoz I have done the required changes and modified the structure of Image Hooks. Please have a look at it and give your suggestions.

Right now I have commented out the store part of EXIF and IPTC because that will only work when the PR of database.xml change is merged. Though, I have tested it and works.

### CODE REVIEW

**Comment by imjalpreet** [comment]
@oparoz I have added extract method for all the three metadata types but for store method to work #23 has to be merged.

File: lib/Hooks/ImageHooks.php [hunk 1]
```diff
@@ -49,23 +51,10 @@ public function register() {
 	public function postCreate(Node $node) {
 		$absolutePath = $this->dataDirectory.$node->getPath();
 
-		$dimensions = getimagesize($absolutePath);
+		$metadataExtracter = new ExtractMetadata($absolutePath);
```
Comment (oparoz) [inline lib/Hooks/ImageHooks.php]: Please use DI

File: lib/Hooks/ImageHooks.php [hunk 2]
```diff
@@ -49,23 +51,10 @@ public function register() {
...
-			$logger->log('debug', 'Image Width: '.$image_width, array('app' => 'MediaMetadata'));
-
-			$imageDimension = new ImageDimension();
-			$imageDimension->setImageId($node->getId());
-			$imageDimension->setImageHeight($image_height);
-			$imageDimension->setImageWidth($image_width);
-
-			$this->mapper->insert($imageDimension);
-		}
+		$dbManager = new StoreMetadata($node, $this->mapper);
```
Comment (oparoz) [inline lib/Hooks/ImageHooks.php]: Please use DI

File: lib/Services/StoreMetadata.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+			$imageDimension->setGpsLongitude($metadata['EXIFData']['gpsLongitude']);
+		}
+*/
+
+	//TODO: Write store methods for IPTC and XMP Metadata
+
+		//Insert to Database
+		$entity = $this->imageDimensionMapper->insert($imageDimension);
+	}
+}
```
Comment (oparoz) [inline lib/Services/StoreMetadata.php]: Please add an empty line

File: lib/Services/ExtractMetadata.php [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+						$c = count($iptc[$s]);
+						for ($i = 0; $i < $c; $i++) {
+							$logger->warning($s . ' = ' . $iptc[$s][$i], array('app' => 'MediaMetadata'));
+						}
+					}
+				}
+			}
+		}
+	}
+}
```
Comment (oparoz) [inline lib/Services/ExtractMetadata.php]: Please add an empty line

File: lib/Hooks/ImageHooks.php [hunk 5]
```diff
@@ -49,23 +51,10 @@ public function register() {
...
+		$dbManager->store($metadata);
```
Comment (oparoz) [inline lib/Hooks/ImageHooks.php]: Please assign a Boolean

**Comment by oparoz** [comment]
OK, I've added some comments in-line. Once those are fixed, you need to add the test for postCreate

File: lib/Hooks/ImageHooks.php (hunk 1 above)
Comment (imjalpreet) [inline lib/Hooks/ImageHooks.php]: @oparoz Can you give a basic idea on how to implement DI over here? I am not able to figure out what the interface will be.

Comment (oparoz) [inline lib/Hooks/ImageHooks.php]: Sure, I've mentioned how it works in the ImageHooks PR I think, but here is the official documentation which might be quicker to understand:
https://docs.nextcloud.org/server/9/developer_manual/app/container.html

**Comment by imjalpreet** [comment]
@oparoz I have used Dependency Injection. Please review it.

**Comment by imjalpreet** [comment]
@oparoz I am not able to figure out why Scrutinizer failed, can you check it once?

**Comment by oparoz** [comment]
Great, thanks. Now you can add the postCreate test, it should be easy this time.

**Comment by imjalpreet** [comment]
@oparoz Should I add the test in this PR only?

**Comment by oparoz** [comment]
Good point, let's keep changes small.
- [x] Remove all non relevant code (EXIF, IPTC, etc.). Keep it handy for another PR
- [ ] Merge this PR
- [ ] Create postCreate test in another PR

And don't worry about Scrutinizer, it's a problem on their end.

**Comment by imjalpreet** [comment]
@oparoz I have removed the non relevant part as of now.

**Comment by oparoz** [comment]
Thank you, now you can create the postCreate PR.

**Pull request merged**